spec/lib/reporting/sp_proofing_events_by_uuid_spec.rb

```diff
@@ -0,0 +1,150 @@
+      expected_csv = [
+        ['Date Range', '2024-12-01 - 2024-12-07'],
+        [
+          'UUID',
+          'Workflow Started',
+          'Documnet Capture Started',
+          'Document Captured',
+          'Selfie Captured',
+          'Document Authentication Passed',
+          'SSN Submitted',
+          'Personal Information Submitted',
+          'Personal Information Verified',
+          'Phone Submitted',
+          'Phone Verified',
+          'Verification Workflow Complete',
+          'Identity Verified for In-Band Users',
+          'Identity Verified for Verify-By-Mail Users',
+          'Identity Verified for Fraud Review Users',
+          'Out-of-Band Verification Pending Seconds',
+          'Agency Handoff Visited',
+          'Agency Handoff Submitted',
+        ],
+        [
+          agency_user_login_uuid,
+          true,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          false,
+          0,
+          false,
+          false,
+        ],
+      ]
+      aggregate_failures do
+        report.as_csv.zip(expected_csv).each do |actual, expected|
+          expect(actual).to eq(expected)
+        end
+      end
+    end
+  end
+
+  describe '#to_csv' do
+    it 'generates a csv' do
+      csv = CSV.parse(report.to_csv, headers: false)
+
+      expected_csv = [
+        ['Date Range', '2024-12-01 - 2024-12-07'],
+        [
+          'UUID',
+          'Workflow Started',
+          'Documnet Capture Started',
+          'Document Captured',
+          'Selfie Captured',
+          'Document Authentication Passed',
+          'SSN Submitted',
+          'Personal Information Submitted',
+          'Personal Information Verified',
+          'Phone Submitted',
+          'Phone Verified',
+          'Verification Workflow Complete',
+          'Identity Verified for In-Band Users',
+          'Identity Verified for Verify-By-Mail Users',
+          'Identity Verified for Fraud Review Users',
+          'Out-of-Band Verification Pending Seconds',
+          'Agency Handoff Visited',
+          'Agency Handoff Submitted',
+        ],
+        [
+          agency_user_login_uuid,
+          'true',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          'false',
+          '0',
+          'false',
+          'false',
+        ],
+      ]
+
+      aggregate_failures do
+        csv.map(&:to_a).zip(expected_csv).each do |actual, expected|
+          expect(actual).to eq(expected)
+        end
+      end
+    end
+  end
+
+  describe '#as_emailable_reports' do
+    it 'returns an emailable report'
+  end
```

Comment:
should we add a body to this spec? or remove it?

Reply:
Whoops! I didn't realize I left this pending. Willfix